### PR TITLE
Enable debug logging when restarting NVDA with add-ons

### PR DESCRIPTION
**Pull request by CyrilleB79**
This PR fixes the initial request of #11538.

The discussion in #11538 and a follow up PR by @mzanm  (#13214) have led to the proposal of a more elaborated solution with a re-design of NVDA's exit dialog. Though this redesign has not seen the light of day, initially due to discussions on the best UX for this dialog and then due to a technical difficulty with the chosen technical design. The result is that 4 year later, the initial and main concern described in #11538 is not solved yet 4 year later. The best is the enemy of the good.

The proposal in this PR is to resolve the main concern initially described in #11538, i.e. addressing the most common use cases. On this basis and depending on feedback after it is merged, the remaining required UX improvements in this dialog can be described and implemented in a subsequent PR if needed.

### Link to issue number:

Fixes the initial request of #11538.

### Summary of the issue:
When a user encounters an issue with NVDA, a dev diagnosing the issue may need to ask them a log. In this situation, the most common use case is to ask a log with add-ons disabled (to eliminate possible add-on interferences) and log level set to debug (to get the maximum information to help debugging). Though, NVDA does not provide a handy way to restart with add-ons disabled and log level set to debug.

### Description of user facing changes

NVDA's exit dialog now provides the 4 following permanent options:
* "Exit" (as before)
* "Restart" (as before)
* "Restart with add-ons disabled and debug logging", replacing "Restart with add-ons disabled". This option may be used to discriminate if a bug comes from NVDA or from add-ons, and in case it comes from NVDA, help a developer to investigate and fix it.
* "Restart with debug logging", just renamed from "Restart with debug logging enabled". This option is useful to get a log to investigate a bug in an add-on.

The non-permanent option to install pending updates remains unchanged.

And the options in secure mode remain unchanged. More specifically, "Restart with add-ons disabled" remains without enabling debug logging because logging is disabled in secure mode.

### Description of development approach
Added the `RESTART_WITH_ADDONS_DISABLED_AND_DEBUG_LOGGING` item in `gui.exit._ExitAction` enum. And removed the unneeded items of this enum to display the combo-box. More specifically, `RESTART_WITH_ADDONS_DISABLED_AND_DEBUG_LOGGING` and `RESTART_WITH_ADDONS_DISABLED` cannot coexist in the allowed values of the combo-box.

Also renamed `RESTART_WITH_DEBUG_LOGGING_ENABLED` to `RESTART_WITH_DEBUG_LOGGING` so that `RESTART_WITH_ADDONS_DISABLED_AND_DEBUG_LOGGING` does not become a longer `RESTART_WITH_ADDONS_DISABLED_AND_DEBUG_LOGGING_ENABLED`; same (and more importantly) for the displayed string. Moreover, technically there is no debug logging that we can enable or disable in NVDA, but a debug level that may be set to "disabled", to "debug" or other intermediate levels.

At last, "Restart with debug logging" is not named "Restart with add-ons enabled and debug logging", because the add-ons can still be disabled all individually in the add-on store.

### Testing strategy:

Manual tests:
* Tried the 4 items of the menu in normal mode and checked if the log and the add-on availability are as expected
* Tried the 3 items of the menu in secure mode and checked that no log is produced and that the add-ons are disabled in the case when it is expected.

No test of Windows Store app (`config.isAppX`) since the Windows Store version of NVDA is not supported anymore.

### Known issues with pull request:
Does not address the dialog redesign discussed in #11538. This can be done in a subsequent PR if needed.
### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Added an option to restart NVDA with add-ons disabled and debug logging enabled, enhancing user control for troubleshooting.
  
- **Documentation**
	- Updated user guide to clarify new exit options, including the addition of debug logging during restarts.

- **Bug Fixes**
	- Improved the naming conventions and logic associated with exit actions for better usability and understanding.

### CODE REVIEW

File: source/gui/exit.py [hunk 1]
```diff
@@ -43,10 +44,16 @@ def _displayStringLabels(self):
...
+			),
+			self.RESTART_WITH_ADDONS_DISABLED: _(
+				# Translators: An option in the combo box to choose exit action.
+				"Restart with add-ons disabled",
+			),
 			# Translators: An option in the combo box to choose exit action.
-			self.RESTART_WITH_ADDONS_DISABLED: _("Restart with add-ons disabled"),
-			# Translators: An option in the combo box to choose exit action.
-			self.RESTART_WITH_DEBUG_LOGGING_ENABLED: _("Restart with debug logging enabled"),
+			self.RESTART_WITH_DEBUG_LOGGING: _("Restart with debug logging"),
```
Comment (SaschaCowley) [inline source/gui/exit.py:56]: I don't think these wording changes actually improve things for users, and may in fact make things less clear.

You said:
> technically there is no debug logging that we can enable or disable in NVDA, but a debug level that may be set to "disabled", to "debug" or other intermediate levels.

That's as may be, but effectively what you're doing when you set a debug level is enabling all debug levels which are less than or equal to that debug level, so we are, in fact, enabling debug logging.

File: user_docs/en/changes.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ The available options are:
 * eSpeak NG has been updated to 1.52-dev commit `961454ff`. (#16775)
   * Added new languages Faroese and Xextan.
 * When using a multi-line braille display via the standard HID braille driver, all lines of cells will be used. (#16993, @alexmoon)
+* The exit dialog now allow to restart NVDA with add-ons disabled and debug logging simultaneously. (#11538, @CyrilleB79)
```
Comment (SaschaCowley) [inline user_docs/en/changes.md:27]:
```suggestion
* The exit dialog now allows you to restart NVDA with add-ons disabled and debug logging enabled simultaneously. (#11538, @CyrilleB79)
```

File: user_docs/en/userGuide.md [hunk 3]
```diff
@@ -1753,7 +1753,7 @@ This option is a checkbox that, when checked, tells NVDA to automatically save t
 ##### Show exit options when exiting NVDA {#GeneralSettingsShowExitOptions}
 
 This option is a checkbox that allows you to choose whether or not a dialog appears when you exit NVDA that asks what action you want to perform.
-When checked, a dialog will appear when you attempt to exit NVDA asking whether you want to exit, restart, restart with add-ons disabled or install pending updates (if any).
+When checked, a dialog will appear when you attempt to exit NVDA, asking whether you want to exit, restart, restart with add-ons disabled and debug logging, restart with debug logging, or install pending updates (if any).
```
Comment (SaschaCowley) [inline user_docs/en/userGuide.md:1756]: This would also require changing if we revert the wording to "debug logging enabled".

**Review by SaschaCowley — CHANGES_REQUESTED** [review]
Thanks Cyrille, this will definitely make users' and developers' jobs easier when attempting to resolve issues. I'm just a bit unsure about the changes in wording.

File: source/gui/exit.py (hunk 1 above)
Comment (CyrilleB79) [reply source/gui/exit.py:56]: OK, I trust you for what sounds better in this case and I understand your reasoning.

Though the first reason why I have changed this in the first place is to be similar with "Restart with add-ons disabled and debug logging" item. So I should change it anyway to "Restart with add-ons disabled and debug logging enabled". But I was concerned that this wording was a bit long in the combo-box. What do you think?

If you do not share the same concerns, I will add "enabled" for both items containing debug logging.

Comment (SaschaCowley) [reply source/gui/exit.py:56]: Hmm, I definitely see what you mean about it being quite long. Probably @seanbudd or a sighted dev would be best to answer that, I tend to only use magnification as a last resort.

Comment (CyrilleB79) [reply source/gui/exit.py:56]: Re the visual aspect, the text is not cropped. The length of the window correctly adapts to its content.

But actually, I have written "quite long" thinking to the combo-box item to be heard (i.e. long for an choice label), not to the visual aspect.

Comment (SaschaCowley) [reply source/gui/exit.py:56]: @CyrilleB79, in conversation with @michaelDCurran, I think "debug logging enabled" is the most understandable wording. Ideally we would rework the exit dialog though, but as you say, that has been stalled for a long time, and the perfect is the enemy of the good.

Comment (CyrilleB79) [reply source/gui/exit.py]: OK, I have restored the original wording "Restart with debug logging enabled" and updated the new wording similarly.

File: user_docs/en/userGuide.md (hunk 3 above)
Comment (CyrilleB79) [reply user_docs/en/userGuide.md]: Done in 4ba25175d7628c511a316aa52a4a9c970257994e

**Review by SaschaCowley — APPROVED** [review]
Thanks Cyrille, this looks good to me now.

File: user_docs/en/changes.md [hunk 4]
```diff
@@ -24,6 +24,7 @@ The available options are:
 * eSpeak NG has been updated to 1.52-dev commit `961454ff`. (#16775)
   * Added new languages Faroese and Xextan.
 * When using a multi-line braille display via the standard HID braille driver, all lines of cells will be used. (#16993, @alexmoon)
+* The exit dialog now allows you to restart NVDA with add-ons disabled and debug logging enabled simultaneously. (#11538, @CyrilleB79)
```
Comment (seanbudd) [inline user_docs/en/changes.md:27]: can this be moved to 2025.1 please

File: user_docs/en/userGuide.md [hunk 5]
```diff
@@ -1753,7 +1753,7 @@ This option is a checkbox that, when checked, tells NVDA to automatically save t
 ##### Show exit options when exiting NVDA {#GeneralSettingsShowExitOptions}
 
 This option is a checkbox that allows you to choose whether or not a dialog appears when you exit NVDA that asks what action you want to perform.
-When checked, a dialog will appear when you attempt to exit NVDA asking whether you want to exit, restart, restart with add-ons disabled or install pending updates (if any).
+When checked, a dialog will appear when you attempt to exit NVDA, asking whether you want to exit, restart, restart with add-ons disabled and debug logging enabled, restart with debug logging enabled, or install pending updates (if any).
```
Comment (seanbudd) [inline user_docs/en/userGuide.md:1756]: this might be better served as a list now. This sentence is quite long

Comment (CyrilleB79) [reply user_docs/en/userGuide.md]: Done

**Review by seanbudd — APPROVED** [review]

**Review by Qchristensen — APPROVED** [review]
This will be really handy, as setting debug logging AND disabling add-ons is something we commonly request of users when troubleshooting, great work!

**Pull request merged**